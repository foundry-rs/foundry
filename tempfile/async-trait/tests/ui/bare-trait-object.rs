#![deny(bare_trait_objects)]

use async_trait::async_trait;

#[async_trait]
trait Trait {
    async fn f(&self);
}

#[async_trait]
impl Trait for Send + Sync {
    async fn f(&self) {}
}

fn main() {}
