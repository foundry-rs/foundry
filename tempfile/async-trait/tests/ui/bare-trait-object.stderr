error[E0782]: expected a type, found a trait
  --> tests/ui/bare-trait-object.rs:11:16
   |
11 | impl Trait for Send + Sync {
   |                ^^^^^^^^^^^
   |
help: you can add the `dyn` keyword if you want a trait object
   |
11 | impl Trait for dyn Send + Sync {
   |                +++
help: alternatively use a blanket implementation to implement `Trait` for all types that also implement `Send + Sync`
   |
11 | impl<T: Send + Sync> Trait for T {
   |     ++++++++++++++++           ~
